File: php/features.php
<?php
declare(strict_types=1);

// MyIp SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class MyIpFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new MyIpBaseFeature();
            case "test":
                return new MyIpTestFeature();
            default:
                return new MyIpBaseFeature();
        }
    }
}

File: py/features.py
# MyIp SDK feature factory

from feature.base_feature import MyIpBaseFeature
from feature.test_feature import MyIpTestFeature


def _make_feature(name):
    features = {
        "base": lambda: MyIpBaseFeature(),
        "test": lambda: MyIpTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
